app/controllers/user_controller.rb

```diff
@@ -0,0 +1,37 @@
+get '/users/new' do
+  erb :new_user
+end
+
+post '/users' do
+  session.clear
+  user = User.new(username: params[:username], email: params[:email], password: params[:password], phone: params[:phone_number])
+  user.save
+  portfolio = Portfolio.new(user_id: user.id, name: params[:portfolio])
+  portfolio.save
+  session[:current_user_id] = user.id
+  session[:current_portfolio_id] = portfolio.id
```

Comment:
you shouldnt store resource ids in the session. Put the resource ids in your urls.